File: 2022/day_01.py
#!/usr/bin/env python3
"""
Day 01:

https://adventofcode.com/2022/day/1
"""
import sys

DAY = "01"
Data = list[list[int]]


def parse_data() -> Data:
    if len(sys.argv) >= 2:
        path = sys.argv[1]
    else:
        path = f"inputs/example_{DAY}.txt"

    with open(path, encoding="utf-8") as f:
        raw = f.read()

    data = [
        [int(line or "0") for line in lines.split("\n")] for lines in raw.split("\n\n")
    ]
    return data


def part_1(data: Data):
    return max(sum(group) for group in data)


def part_2(data: Data):
    return sum(sorted((sum(group) for group in data), reverse=True)[:3])


def main():
    data = parse_data()

    print(part_1(data))
    print(part_2(data))


if __name__ == "__main__":
    main()

File: 2022/day_01.rb
#!/usr/bin/env ruby

PATH = ARGV[0] || File.join(__dir__, "inputs", "example_01.txt")

sums_of_calories = File.read(PATH)
  .split("\n\n")
  .map { _1.split("\n").map(&:to_i).sum }

def part_1(data)
  data.max
end

def part_2(data)
  data.max(3).sum
end

puts part_1(sums_of_calories)
puts part_2(sums_of_calories)
